dbms/src/Common/Spiller.cpp

```diff
@@ -0,0 +1,148 @@
+// Copyright 2023 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/Spiller.h>
+#include <DataStreams/NativeBlockOutputStream.h>
+#include <DataStreams/NullBlockInputStream.h>
+#include <DataStreams/SpilledFilesInputStream.h>
+#include <IO/CompressedWriteBuffer.h>
+#include <IO/WriteBufferFromFile.h>
+#include <Poco/Path.h>
+
+
+namespace DB
+{
+SpilledFile::SpilledFile(const String & file_name_)
+    : Poco::File(file_name_)
+{}
+
+SpilledFile::~SpilledFile()
+{
+    try
+    {
+        if (exists())
+            remove(true);
+    }
+    catch (...)
+    {
```

Comment:
add some log here?
This way, if there is a problem with cleaning, we can know where the files are not being cleaned and clean them manually

Reply:
OK